core/trino-spi/src/main/java/io/trino/spi/connector/ConnectorTableLayout.java

```diff
@@ -22,11 +22,20 @@
 {
     private final Optional<ConnectorPartitioningHandle> partitioning;
     private final List<String> partitionColumns;
+    private final boolean multipleWritersPerPartitionSupported;
 
     public ConnectorTableLayout(ConnectorPartitioningHandle partitioning, List<String> partitionColumns)
+    {
+        // Keep the value of multipleWritersPerPartitionSupported false by default if partitioning is present
+        // for backward compatibility.
+        this(partitioning, partitionColumns, false);
+    }
+
+    public ConnectorTableLayout(ConnectorPartitioningHandle partitioning, List<String> partitionColumns, boolean multipleWritersPerPartitionSupported)
```

Comment:
nit: this method would most likely be used with `multipleWritersPerPartitionSupported=true`. We could consider switching to static constructor methods in the future